modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsaMPCv2.ts

```diff
@@ -0,0 +1,485 @@
+import assert from 'assert';
+import { NonEmptyString } from 'io-ts-types';
+import { DklsDkg, DklsTypes, DklsComms } from '@bitgo/sdk-lib-mpc';
+
+import { Keychain, KeyType } from '../../../keychain';
+import { KeychainsTriplet } from '../../../baseCoin';
+import { generateGPGKeyPair } from '../../opengpgUtils';
+import { BaseEcdsaUtils } from './base';
+import {
+  MPCv2Party,
+  MPCv2KeyGenState,
+  MPCv2BroadcastMessage,
+  MPCv2KeyGenRound1Response,
+  MPCv2KeyGenRound2Response,
+  MPCv2KeyGenRound3Response,
+  MPCv2P2PMessage,
+  KeyGenTypeEnum,
+  MPCv2KeyGenStateEnum,
+} from '@bitgo/public-types';
+import { GenerateMPCv2KeyRequestBody, GenerateMPCv2KeyRequestResponse, MPCv2PartiesEnum } from './typesMPCv2';
+
+export class EcdsaMPCv2Utils extends BaseEcdsaUtils {
+  /** @inheritdoc */
+  async createKeychains(params: {
+    passphrase: string;
+    enterprise: string;
+    originalPasscodeEncryptionCode?: string;
+  }): Promise<KeychainsTriplet> {
+    const m = 2;
+    const n = 3;
+    const userSession = new DklsDkg.Dkg(n, m, MPCv2PartiesEnum.USER);
+    const backupSession = new DklsDkg.Dkg(n, m, MPCv2PartiesEnum.BACKUP);
+    const userGpgKey = await generateGPGKeyPair('secp256k1');
+    const backupGpgKey = await generateGPGKeyPair('secp256k1');
+
+    // Get the BitGo public key based on user/enterprise feature flags
+    // If it doesn't work, use the default public key from the constants
+    const bitgoPublicGpgKey = (
+      (await this.getBitgoGpgPubkeyBasedOnFeatureFlags(params.enterprise)) ?? this.bitgoPublicGpgKey
```

Comment:
Can we switch to having DKLs using a baked in BitGo pub key instead of fetching from a server? It is much more secure that way

Reply:
Out of scope for now. We can do this before GA.